### ISSUE TITLE
Incorrect classification of changed comments

### ISSUE THREAD

**Issue by husttroy** [julioleitao/codereview-file-classifier#4]
If a comment is in the same line with a program statement, a change on the comment is also identified as a change on the program statement, even though the program statement is not changed at all.

Check java/src/jmri/jmrix/loconet/LnOpsModeProgrammer.java in [this PR](https://github.com/JMRI/JMRI/pull/5345/files). 

![issue-3](https://user-images.githubusercontent.com/2655256/41751124-b03494da-7574-11e8-873f-c9b6918b0616.png)

**Comment by julioleitao** [comment]
Thanks for this feedback, @husttroy!

Yes, this is a problem I also saw yesterday. This is a problem on how we can measure the semantic change at the AST tree between the diff code. 

I made a simple check to verify if a node of the AST has changed at the diff line. This was because another approach contains a limited detection or issues. 

I am going to investigate how to better detect and resolve this issue.
